service/controller/resource/values/create.go

```diff
@@ -17,10 +18,18 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
+	if app.GetDeletionTimestamp() != nil {
+		r.logger.Debugf(ctx, "App CR %q is has deletion timestamp", app.Name)
+		r.logger.Debugf(ctx, "cancelling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
+		return nil
+	}
+
 	configVersion, ok := app.GetAnnotations()[annotation.ConfigVersion]
 	if !ok {
 		r.logger.Debugf(ctx, "App CR %q is missing %q annotation", app.Name, annotation.ConfigVersion)
 		r.logger.Debugf(ctx, "cancelling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
`resourcecanceledcontext` should be removed. This was useful only with obsolete CRUD resources.